### PR TITLE
(feat): improved accessibility for modal box

### LINKED ISSUE #5912: [FEATURE] Improve accesibility for the Modal Box
### Description

I have shared the screenshot of the page, I request the feature accessibility of the Modal box should be improved. 

1. When we click ESC then it should close the modal but is not closing it
2. When we click outside of Modal box it should close it is not.

Visit: https://linkfree.eddiehub.io/playground

### Screenshots

Screenshot 2023-04-08 at 11 21 44 AM

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by NiazMorshed2007**
### This pr improves the modal accessibility as discussed in #5912 

fixes #5912 

#### What feature I added?
1.  modal closes when pressed `esc` on keyboard
2. modal closes if clicks outside the modal box

### `1. Close modal when esc is pressed on keyboard`
In order to add this feature I approached with listening to `esc` keystroke on document. And thus the `esc` is pressed it calls the `toggle()` methods which is passed in the `props` of `ProfilePreview` component to close the modal.

##### Code for this feature 👇
``` js
 useEffect(() => {
    const handleEscapeKeyPress = (event) => {
      if (event.key === "Escape") {
        toggle();
      }
    }

    document.addEventListener("keydown", handleEscapeKeyPress);

    return () => {
      document.removeEventListener("keydown", handleEscapeKeyPress);
    };
  }, []);
  ```
  
  ### `2. Close modal when click outside the modal box`
  To figure out whether user clicked in the modal box or outside the modal box, I took a reference of modal box with `useRef` and listening to `onClick` function on the modal wrapper. And on the `onClick` function I'm checking if the reference of the modalBox contains the click event. If it not contains it will run the `toggle()` function to close the modal.

##### Code for this feature 👇
```js

  const modalContentRef = useRef(null)

  const handleClickOutsideContentBox = (event) => {
    if (modalContentRef.current && !modalContentRef.current.contains(event.target)) {
      toggle();
    }
  }

return <div onClick={handleClickOutsideContentBox}}> // onClick function on modal wrapper
<div ref={modalContentRef}>...</div> // modal content box
</div>
```

That's the wrap :)
Thanks!

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/5932"></a>

### CODE REVIEW

File: package-lock.json
Comment (ThePiyushAggarwal) [inline package-lock.json:1]: Please remove this file from the PR

Comment (NiazMorshed2007) [reply package-lock.json:1]: hey! thanks for the review. I'm having some issues to remove it from the pr. So I changed it to draft, can you change it?

Comment (ThePiyushAggarwal) [reply package-lock.json:1]: Download the `package-lock.json` from the master branch and then paste in your local machine and commit. I think that should work.

File: components/modals/ProfilePreview.js [hunk 1]
```diff
@@ -1,18 +1,42 @@
...
+  useEffect(() => {
+    const handleEscapeKeyPress = (event) => {
+      if (event.key === "Escape") {
+        toggle();
+      }
+    }
+
+    document.addEventListener("keydown", handleEscapeKeyPress);
+
+    return () => {
```
Comment (eddiejaoude) [inline components/modals/ProfilePreview.js:24]: Please can you check your VSCode is using our coding standards, I think you might need the Prettier plugin

File: components/modals/ProfilePreview.js [hunk 2]
```diff
@@ -1,18 +1,42 @@
 import UserPage from "@components/user/UserPage";
+import { useEffect, useRef } from "react";
 
 export default function Preview({ toggle, data }) {
   const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL;
+  const modalContentRef = useRef(null)
+
+
```
Comment (eddiejaoude) [inline components/modals/ProfilePreview.js:8]: Thank you for the contribution, I don't think 2 blank lines are needed here

File: components/modals/ProfilePreview.js (hunk 1 above)
Comment (NiazMorshed2007) [reply components/modals/ProfilePreview.js:24]: Sure, I'm making the changes

File: components/modals/ProfilePreview.js (hunk 2 above)
Comment (NiazMorshed2007) [reply components/modals/ProfilePreview.js:8]: Noted

File: package-lock.json
Comment (eddiejaoude) [reply package-lock.json:1]: This YouTube video I created will show why the file got changed and how to fix it
https://www.youtube.com/watch?v=jHkIXYVpMTo

Comment (NiazMorshed2007) [reply package-lock.json:1]: @eddiejaoude Thanks mate! This is the thing I was looking for. Learned something new :D

**Review by eddiejaoude — APPROVED** [review]
Looks great 👍  thank you!

**Pull request merged**